baseplate/server/runtime_monitor.py

```diff
@@ -0,0 +1,168 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import gc
+import logging
+import os
+import socket
+import threading
+import time
+
+from baseplate import config
+
+
+REPORT_INTERVAL_SECONDS = 10
+
+
+class _ConcurrencyReporter(object):
+    def __init__(self, pool):
+        self.pool = pool
+
+    def report(self, batch):
+        batch.gauge("active_requests").replace(len(self.pool.greenlets))
+
+
+class _BlockedGeventHubReporter(object):
+    def __init__(self, max_blocking_time):
+        try:
+            import gevent.events
+        except ImportError:
+            raise Exception("Gevent >=1.3 required")
+
+        gevent.events.subscribers.append(self._on_gevent_event)
+        gevent.config.monitor_thread = True
+        gevent.config.max_blocking_time = max_blocking_time
+        gevent.get_hub().start_periodic_monitoring_thread()
+
+        self.times_blocked = []
+
+    def _on_gevent_event(self, event):
+        import gevent.events
+
+        if isinstance(event, gevent.events.EventLoopBlocked):
+            self.times_blocked.append(event.blocking_time)
```

Comment:
do you know what's in `event.info` and whether that would be useful to see?

Reply:
It's a list of strings, each one being a line of the thing it outputs to the logs. This is the stack trace of the offending greenlet first followed by the stack traces of all other active greenlets. I think it might be interesting to get at least the first stack trace into sentry, for sure, but at minimum we'll get that data in our logs.

http://www.gevent.org/api/gevent.events.html#gevent.events.IEventLoopBlocked